core-bundle/src/Resources/contao/library/Contao/Controller.php

```diff
@@ -422,6 +423,21 @@ public static function getArticle($varId, $blnMultiMode=false, $blnIsInsertTag=f
 		return $strBuffer;
 	}
 
+	/**
+	 * Generate an article and return it as string
+	 *
+	 * @param mixed   $intId          The article ID
+	 * @param boolean $blnMultiMode   If true, only teasers will be shown
+	 * @param boolean $blnIsInsertTag If true, there will be no page relation and no HTML markup for the container
+	 * @param string  $strColumn      The name of the column
+	 *
+	 * @return string|boolean The article HTML markup or false
+	 */
+	public static function getArticleById($intId, $blnMultiMode=false, $blnIsInsertTag=false, $strColumn='main')
```

Comment:
I think only the `$intId` parameter should be used here, without the other three parameters.

Reply:
But then you will not be able to control whether you want the container markup. The `$strColumn` variable on the other hand seems to be a left over from the past. It is passed to the `ModuleArticle` instance which in turn passes it to the template. But the current default `mod_article` template does not use this variable.

Reply:
Then why did you add it to the new method?

Reply:
Ah, it is used here:

https://github.com/contao/contao/blob/26b355525b045fedf129f55e48253d75c8cedb91/core-bundle/src/Resources/contao/modules/ModuleArticle.php#L193

Reply:
> Then why did you add it to the new method?

I only looked at it when @ausi questioned the addition of all parameters ;)



> Ah, it is used here:

Hm, same story there though. It is passed to the template, but not used by default anywhere. But some people might have custom template and check on the column.

Reply:
I think you have to search for "inColumn" to find its usage, e.g. here:

https://github.com/contao/contao/blob/b48f650dd49a58daae5883ded66aac311ccffd75/core-bundle/src/Resources/contao/elements/ContentTeaser.php#L75-L78